packages/organizations/origin-organization-irec-api/src/beneficiary/handlers/add-organization-beneficiary.handler.ts

```diff
@@ -0,0 +1,43 @@
+import { Repository } from 'typeorm';
+import { CommandBus, CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { AddOrganizationBeneficiaryCommand, GetBeneficiaryCommand } from '../commands';
+import { Beneficiary } from '../beneficiary.entity';
+import { NotFoundException } from '@nestjs/common';
+import { BeneficiaryDTO } from '../dto/beneficiary.dto';
+
+@CommandHandler(AddOrganizationBeneficiaryCommand)
+export class AddOrganizationBeneficiaryHandler
+    implements ICommandHandler<AddOrganizationBeneficiaryCommand>
+{
+    constructor(
+        @InjectRepository(Beneficiary)
+        private readonly repository: Repository<Beneficiary>,
+        private readonly commandBus: CommandBus
+    ) {}
+
+    public async execute({
+        ownerId,
+        irecBeneficiaryId
+    }: AddOrganizationBeneficiaryCommand): Promise<BeneficiaryDTO> {
+        const beneficiary = await this.repository.findOne({
+            ownerId: null,
+            irecBeneficiaryId
+        });
```

Comment:
In order to add beneficiaries to your organization list, it has to exist first in the DB (with ownerId=null), what's the process of filling that?

Reply:
1. The user gets all the public beneficiaries (with ownerId=null)
2. The user adds beneficiaries to his organization's private beneficiary list. (to use later with certificates redemption)
3. When the beneficiary is not needed anymore, it is deleted from the private beneficiaries list by `DELETE /irec/beneficiary/:id`

Reply:
What's the process of getting a public list of beneficiaries?

Reply:
`GET /irec/beneficiary` returns list of public beneficiaries
`GET /irec/beneficiary/company` returns list of private ones